activerecord/lib/active_record/connection_adapters/postgresql/oid/range.rb

```diff
@@ -69,13 +69,30 @@ def type_cast_single_for_database(value)
             def extract_bounds(value)
               from, to = value[1..-2].split(",", 2)
               {
-                from:          (from == "" || from == "-infinity") ? infinity(negative: true) : from,
-                to:            (to == "" || to == "infinity") ? infinity : to,
+                from:          (from == "" || from == "-infinity") ? infinity(negative: true) : unquote(from),
+                to:            (to == "" || to == "infinity") ? infinity : unquote(to),
                 exclude_start: value.start_with?("("),
                 exclude_end:   value.end_with?(")")
               }
             end
 
+            # When formatting the bound values of range types, PostgreSQL quotes
+            # the bound value using double-quotes in certain conditions (manual
+            # sections 8.17.5, 8.16.6). Within a double-quoted string, literal "
+            # and \ characters are themselves escaped. In input, PostgreSQL
+            # accepts multiple escape styles for " (either \" or "") but in
+            # output always uses "" (see 8.16.6).
```

Comment:
Can you add the link for the doc?

Example:

https://github.com/rails/rails/blob/55f519f5b62db017077e2fc8aef31ba31708e038/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L296-L297
https://github.com/rails/rails/blob/1944a7e74c6c1b7a6234414a00d294412c05fde1/activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb#L219-L221
https://github.com/rails/rails/blob/5fdf76f8fd585bdd8ba6a5d9a6132b673b8b205f/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L69-L74

Reply:
Certainly!

Reply:
Also, to be honest I'd not prefer to mention about dynamic section number, actually section number is a little fragile.

Example:

![image](https://user-images.githubusercontent.com/12642/84340746-54146900-abdc-11ea-8d07-6bdff3626cff.png)

https://www.postgresql.org/docs/12/libpq-envars.html

![image](https://user-images.githubusercontent.com/12642/84340767-6393b200-abdc-11ea-9c75-eca43ef00699.png)

https://www.postgresql.org/docs/11/libpq-envars.html

Reply:
Sure, happy to make that change